Source/MQTTnet/Internal/Disposable.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+
+namespace MQTTnet.Internal
+{
+    public class Disposable : IDisposable
+    {
+        protected bool IsDisposed => _isDisposed;
+
+        protected void ThrowIfDisposed()
+        {
+            if (_isDisposed)
+            {
+                throw new ObjectDisposedException(GetType().Name);
+            }
+        }
+
+
+        #region IDisposable Support
```

Comment:
this is the idisposable implementation Visual studio generates i left the TODOs in place so anyone has a cheat sheet where to put the code